packages/svelte/src/compiler/phases/2-analyze/index.js

```diff
@@ -1252,7 +1253,12 @@ const common_visitors = {
 							!should_proxy_or_freeze(binding.initial.arguments[0], context.state.scope)))) ||
 					binding.kind === 'frozen_state' ||
 					binding.kind === 'derived') &&
-				context.state.function_depth === binding.scope.function_depth
+				// We're only concerned with reads here
+				parent.type !== 'AssignmentExpression' &&
+				parent.type !== 'UpdateExpression' &&
+				(parent.type !== 'MemberExpression' ||
+					(context.path.at(-2)?.type !== 'AssignmentExpression' &&
+						context.path.at(-2)?.type !== 'UpdateExpression'))
```

Comment:
what case is this covering? member expressions already don't trigger the warning IIUC

Reply:
You're right, this is unnecessary

Reply:
Wait I also need to check it's the left hand side of the assignment